tests/firedrake/regression/test_fieldsplit_cofunction.py

```diff
@@ -0,0 +1,60 @@
+import firedrake as fd
+
+
+def test_fieldsplit_cofunction():
```

Comment:
Was there an already existing test for Fieldsplit that used a form RHS that could be extended to test Cofunctions?

Reply:
Sure, that's a better idea. This was just making the script I was bugfixing with into a test. I'll find a suitable existing one and parameterise it.

Reply:
I parameterised a couple extra tests with cofunction rhs because the one in `test_matrix_free` just solves a mass matrix so the form is block diagonal.
The two extra tests have off-diagonal blocks in the form so will make sure that we DTRT in that case.